pkg/core/interop_neo.go

```diff
@@ -138,6 +138,8 @@ func contractUpdate(ic *interop.Context, v *vm.VM) error {
 		if len(siMap) != 0 {
 			return errors.New("old contract shouldn't have storage")
 		}
+	} else if !newcontract.ScriptHash().Equals(contract.ScriptHash()) {
```

Comment:
I'm now thinking about the necessity of this check. We have the same one just 20 lines above in case if the script of the new contract isn't' nil, and it returns an error if so. It's technically possible that new script is nil, but shouldn't we return an error in this case (although C# does not)? It seemed to me strange that we can process contracts without script.

Or if the original C# idea was to provide either [new script] or [new manifest] or [new script and manifest] and update only provided values, we don't have matching behavior, because `createContractStateFromVM` assumes that manifest is required (and C# do not, they have two checks for null: for script and manifest, see https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Contract.cs#L78).

So, could you, please, clarify the way this method should work?